### ISSUE TITLE
Compatibility issue ansible-lint 6.17.0

### ISSUE THREAD

**Issue by mac-roche** [ansible-community/molecule#3927]
# Issue Type

- Bug report

# Molecule and Ansible details

```
ansible --version && molecule --version
ansible [core 2.15.0]
  config file = /home/USERID/.ansible.cfg
  configured module search path = ['/home/USERID/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/USERID/.pyenv/versions/spot311/lib/python3.11/site-packages/ansible
  ansible collection location = /home/USERID/.ansible/collections:/usr/share/ansible/collections
  executable location = /home/USERID/.pyenv/versions/spot311/bin/ansible
  python version = 3.11.2 (main, May  9 2023, 12:34:19) [GCC 11.3.0] (/home/USERID/.pyenv/versions/spot311/bin/python)
  jinja version = 3.1.2
  libyaml = True
molecule 5.0.1 using python 3.11 
    ansible:2.15.0
    azure:23.4.1 from molecule_plugins
    containers:23.4.1 from molecule_plugins requiring collections: ansible.posix>=1.3.0 community.docker>=1.9.1 containers.podman>=1.8.1
    delegated:5.0.1 from molecule
    docker:23.4.1 from molecule_plugins requiring collections: community.docker>=3.0.2 ansible.posix>=1.4.0
    ec2:23.4.1 from molecule_plugins
    gce:23.4.1 from molecule_plugins requiring collections: google.cloud>=1.0.2 community.crypto>=1.8.0
    podman:23.4.1 from molecule_plugins requiring collections: containers.podman>=1.7.0 ansible.posix>=1.3.0
    vagrant:23.4.1 from molecule_plugins
```

Molecule installation method (one of):

- pip

Ansible installation method (one of):

- pip

Detail any linters or test runners used:

# Desired Behavior

I want to update ansible-lint to the latest version (6.17.0):

```
pip install --upgrade ansible-lint
```

# Actual Behaviour

Updating ansible-lint results in a compatibility error with ansible-compat caused by molecule:

```
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
molecule 5.0.1 requires ansible-compat<4.0.0,>=2.2.0, but you have ansible-compat 4.1.2 which is incompatible.
Successfully installed ansible-compat-4.1.2
```

**Comment by ziegenberg** [comment]
This will be fixed with #3904 and it's probably a related duplicate of #3903.

**Comment by ssbarnea** [comment]
Fixed by https://github.com/ansible-community/molecule/discussions/3942
